packages/dusk-cli/src/lib/validateGameFiles.ts

```diff
@@ -257,85 +281,64 @@ async function validateMultiplayerLogicJsContent({
+      !gameConfig.minPlayers ||
+      typeof gameConfig.minPlayers !== "number" ||
+      isNaN(gameConfig.minPlayers)
     ) {
       errors.push({
         message: "logic.js: minPlayers not found or is invalid",
       })
-    }
-
-    if (
-      !multiplayerMetadata.maxPlayers ||
-      isNaN(multiplayerMetadata.maxPlayers)
+    } else if (
+      gameConfig.minPlayers &&
+      (gameConfig.minPlayers < 1 || gameConfig.minPlayers > MAX_PLAYERS)
     ) {
       errors.push({
-        message: "logic.js: maxPlayers not found or is invalid",
+        message: `logic.js: minPlayers must be between 1 and ${MAX_PLAYERS}`,
       })
     }
 
     if (
-      multiplayerMetadata.minPlayers &&
-      (multiplayerMetadata.minPlayers < 1 ||
-        multiplayerMetadata.minPlayers > MAX_PLAYERS)
+      !gameConfig.maxPlayers ||
+      typeof gameConfig.maxPlayers !== "number" ||
+      isNaN(gameConfig.maxPlayers)
     ) {
       errors.push({
-        message: `logic.js: minPlayers must be between 1 and ${MAX_PLAYERS}`,
+        message: "logic.js: maxPlayers not found or is invalid",
       })
-    }
-
-    if (
-      multiplayerMetadata.maxPlayers &&
-      (multiplayerMetadata.maxPlayers < 1 ||
-        multiplayerMetadata.maxPlayers > MAX_PLAYERS)
+    } else if (
+      gameConfig.maxPlayers &&
+      (gameConfig.maxPlayers < 1 || gameConfig.maxPlayers > MAX_PLAYERS)
     ) {
       errors.push({
         message: `logic.js: maxPlayers must be between 1 and ${MAX_PLAYERS}`,
       })
     }
 
     if (
-      multiplayerMetadata.maxPlayers &&
-      multiplayerMetadata.minPlayers &&
-      multiplayerMetadata.maxPlayers < multiplayerMetadata.minPlayers
+      gameConfig.maxPlayers &&
+      gameConfig.minPlayers &&
+      gameConfig.maxPlayers < gameConfig.minPlayers
     ) {
       errors.push({
         message:
           "logic.js: maxPlayers must be greater than or equal to minPlayers",
       })
     }
 
-    if (multiplayerMetadata.updatesPerSecondDefined) {
-      if (multiplayerMetadata.updatesPerSecond === undefined) {
-        errors.push({
-          message:
-            "logic.js: updatesPerSecond must be a constant (updatesPerSecond: 1-30)",
-        })
-      } else if (
-        multiplayerMetadata.updatesPerSecond < 1 ||
-        multiplayerMetadata.updatesPerSecond > 30
-      ) {
-        errors.push({
-          message:
-            "logic.js: updatesPerSecond must be undefined or between 1 and 30",
-        })
-      }
-    }
-
-    if (multiplayerMetadata.inputDelay !== undefined) {
-      if (
-        multiplayerMetadata.inputDelay < 0 ||
-        multiplayerMetadata.inputDelay > 50
-      ) {
-        errors.push({
-          message: "logic.js: inputDelay must be undefined or between 0 and 50",
-        })
-      }
+    if (gameConfig.updatesPerSecond === undefined) {
+      errors.push({
+        message:
+          "logic.js: updatesPerSecond must be a constant (updatesPerSecond: 1-30)",
```

Comment:
This isn't an error case now is it - undefined just means they didn't specify updatesPerSecond. I think it might be worth adding a check for have they specified an update function and no updatesPerSecond or vice versa.

Reply:
Updates per second is always going to be provided, even if dev does not set it to any value.
So theoretically this check is redundant now, but keeping it in case we change something in the future.